sdk/core/core-rest-pipeline/test/auxiliaryAuthenticationPolicy.spec.ts

```diff
@@ -0,0 +1,244 @@
+
+    assert(
+      fakeGetToken.calledWith(tokenScopes, {
+        abortSignal: undefined,
+        tracingOptions: undefined,
+      }),
+      "fakeGetToken called incorrectly."
+    );
+    assert.strictEqual(request.headers.get("x-ms-authorization-auxiliary"), `Bearer ${mockToken}`);
+  });
+
+  it("correctly adds an Authentication header with two access tokens", async function () {
+    const mockToken1 = "token1",
+      mockToken2 = "token2";
+    const tokenScopes = ["scope1", "scope2"];
+    const fakeGetToken1 = sinon.fake.returns(
+      Promise.resolve({ token: mockToken1, expiresOn: new Date() })
+    );
+    const fakeGetToken2 = sinon.fake.returns(
+      Promise.resolve({ token: mockToken2, expiresOn: new Date() })
+    );
+    const mockCredential1: TokenCredential = {
+      getToken: fakeGetToken1,
+    };
+    const mockCredential2: TokenCredential = {
+      getToken: fakeGetToken2,
+    };
+
+    const request = createPipelineRequest({ url: "https://example.com" });
+    const successResponse: PipelineResponse = {
+      headers: createHttpHeaders(),
+      request,
+      status: 200,
+    };
+    const next = sinon.stub<Parameters<SendRequest>, ReturnType<SendRequest>>();
+    next.resolves(successResponse);
+
+    const auxiTokenAuthPolicy = createAuxiliaryTokenPolicy(tokenScopes, [
+      mockCredential1,
+      mockCredential2,
+    ]);
+    await auxiTokenAuthPolicy.sendRequest(request, next);
+
+    assert(
+      fakeGetToken1.calledWith(tokenScopes, {
+        abortSignal: undefined,
+        tracingOptions: undefined,
+      }),
+      "fakeGetToken called incorrectly."
+    );
+    assert.strictEqual(
+      request.headers.get("x-ms-authorization-auxiliary"),
+      `Bearer ${mockToken1},Bearer ${mockToken2}`
+    );
+  });
+
+  it("only refreshes invalid token during the refresh window", async () => {
+    const expireDelayMs = defaultRefreshWindow + 5000;
+    let tokenExpiration = Date.now() + expireDelayMs;
+    const shortCredential = new MockRefreshAzureCredential(tokenExpiration);
+    const longCredential = new MockRefreshAzureCredential(Date.now() + expireDelayMs * 3);
+
+    const request = createPipelineRequest({ url: "https://example.com" });
+    const successResponse: PipelineResponse = {
+      headers: createHttpHeaders(),
+      request,
+      status: 200,
+    };
+    const next = sinon.stub<Parameters<SendRequest>, ReturnType<SendRequest>>();
+    next.resolves(successResponse);
+
+    const policy = createAuxiliaryTokenPolicy("test-scope", [shortCredential, longCredential]);
+
+    // The token is cached and remains cached for a bit.
+    await policy.sendRequest(request, next);
+    await policy.sendRequest(request, next);
+    assert.strictEqual(shortCredential.authCount, 1);
+    assert.strictEqual(longCredential.authCount, 1);
+    assert.strictEqual(
+      request.headers.get("x-ms-authorization-auxiliary"),
+      `Bearer mock-token,Bearer mock-token`
```

Comment:
the docs page has a space in addition to a comma between tokens, not sure if that is important or not

Reply:
I think both could work.